fc/build-from-security.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+import apt
+import atexit
+import glob
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+arch_to_base_uri = {
+    "amd64": "http://archive.ubuntu.com/ubuntu",
+    "i386":  "http://archive.ubuntu.com/ubuntu",
+    # ports
+    "s390x": "http://ports.ubuntu.com/ubuntu-ports",
+    "arm64": "http://ports.ubuntu.com/ubuntu-ports",
+    "armhf": "http://ports.ubuntu.com/ubuntu-ports",
+    "ppc64el": "http://ports.ubuntu.com/ubuntu-ports",
+}
+
+def fetch_source_from_security(release, pkgname):
+    apt.apt_pkg.config.set("Acquire::AllowInsecureRepositories", "1")
+    with tempfile.TemporaryDirectory(prefix="aptroot-{}".format(release)) as base:
+        sources_list = base+"/etc/apt/sources.list"
+        os.makedirs(os.path.dirname(sources_list), exist_ok=True)
+        base_uri = arch_to_base_uri[apt.apt_pkg.get_architectures()[0]]
+        with open(sources_list, "w") as fp:
+            fp.write("""
+deb {base_uri} {dist} main
+deb-src {base_uri} {dist} main
+deb {base_uri} {dist}-security main
+deb-src {base_uri} {dist}-security main
+        """.format(base_uri=base_uri, dist=release))
+
+        cache = apt.Cache(rootdir=base)
+        cache.update(apt.progress.text.AcquireProgress())
+        cache.open()
+        # extract source
+        pkg_src_dir = tempfile.mkdtemp(prefix="{}-{}-src".format(release, pkgname))
+        os.makedirs(pkg_src_dir, exist_ok=True)
+        pkg = cache["fontconfig"]
+        if not glob.glob("{}/{}-*".format(pkg_src_dir, pkgname)):
+            pkg.candidate.fetch_source(destdir=pkg_src_dir)
```

Comment:
I'm unfamiliar with this function-- will it unpack the sources and apply the patches like 'dpkg-source -x' will?

Reply:
Yeah, it will internally call dpkg-source -x - it's similar to "apt-get source".